### ISSUE TITLE
Chrome Extension Won't Connect to Database

### ISSUE THREAD

**Issue by vozeldr** [keepassxreboot/keepassxc-browser#1483]
## Expected Behavior

The chrome extension should connect to the database.

## Current Behavior

The chrome extension will not connect to the database.

## Background

I had to reformat my machine a month or so ago. Upon setting everything back up, the Firefox extension worked perfectly but the Chrome extension was not even listed in the store. I checked back a few days ago and the Chrome extension was back in the store, so I installed it. A few different things in my setup from before my reformat are:

- I installed the flatpak version of KeePassXC
- I started using multiple profiles in Chrome to separate out sites for the 2 companies I do work for.

## Steps to Reproduce (for bugs)

1. On Pop!_OS 21.04, install the flatpak version of KeePassXC from the Pop!_Shop
2. Install Google Chrome 96.0.4664.45
3. Install Firefox 94.0
4. Install KeePassXC-Browser in both Chrome and Firefox
5. Open a database in KeePassXC
6. Ensure Browser Integration is enabled for both Google Chrome and Firefox (I actually tried checking all of them just in case, but no luck)
7. Connect the Firefox extension to the database
8. Attempt to connect the Chrome extension to the database

## Debug info

KeePassXC - 2.6.6
KeePassXC-Browser - 1.7.10.1
Operating system: Linux
Browser: Chrome

```text
Denying load of chrome-extension://oboonakemofpalcgghocfoadofidjkkk/css/notification.css. Resources must be listed in the web_accessible_resources manifest key in order to be loaded by pages outside the extension.

ui.js:235 GET chrome-extension://invalid/ net::ERR_FAILED
kpxcUI.createNotification @ ui.js:235
iconClicked @ username-field.js:122
await in iconClicked (async)
(anonymous) @ username-field.js:95
```

![image](https://user-images.githubusercontent.com/2746303/144432795-5fedd5c2-84fe-4e1f-89a1-725e5b8b3660.png)

![image](https://user-images.githubusercontent.com/2746303/144432887-c7c1f461-ed12-4979-8636-5001d206c741.png)

![image](https://user-images.githubusercontent.com/2746303/144434291-1542c678-36f9-4082-a940-686e1c228128.png)

![image](https://user-images.githubusercontent.com/2746303/144434539-cf01527c-d602-4695-86d0-18d0cd5d7beb.png)

**Comment by varjolintu** [comment]
I'm aware of the notification issue. Does making the connection work from the extension popup?

**Comment by vozeldr** [comment]
Nope

![image](https://user-images.githubusercontent.com/2746303/144436524-091448e1-78ec-4cbe-932a-89ae81124fad.png)

![image](https://user-images.githubusercontent.com/2746303/144436470-ed931350-2180-4713-af0e-c2e3aece633e.png)

**Comment by varjolintu** [comment]
Please check the [Troubleshooting guide](https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide).

**Comment by vozeldr** [comment]
I am using the .deb version of Chrome (not a snap or flatpak).

I looked in `~/.config/google-chrome/NativeMessagingHosts` and there were no files located there. I copied over the json file from the `~/.mozilla/native-messaging-hosts` and edited it as follows

```json
{
    "allowed_origins": [
        "chrome-extension://oboonakemofpalcgghocfoadofidjkkk/"
    ],
    "description": "KeePassXC integration with native messaging support",
    "name": "org.keepassxc.keepassxc_browser",
    "path": "/home/me/.local/share/flatpak/exports/bin/org.keepassxc.KeePassXC",
    "type": "stdio"
}
```

I verified the symlink exists at the location specified in the path and it works. It appears maybe the issue lies in the flatpak version of KeePassXC not correctly installing the json file. I did have to connect to the database separately for each of the chrome profiles, but maybe that's just how it works.
